app/models/flowchart.rb

```diff
@@ -1,4 +1,43 @@
 # frozen_string_literal: true
 
 class Flowchart < ApplicationRecord
+  validates :title, format: { with: /[a-zA-Z ]/ }, null: false
+  validates :description, format: { with: /[a-zA-Z ]/ }, null: false
+  validates :height, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, null: false
+  validates :deleted, inclusion: { in: [true, false] }, null: false
+
+  def calculate_and_set_max_height
+    flowchartnodes = FlowchartNode.where(flowchart_id: id, deleted: false)
+    nodes_indexed_by_id = {}
+    root_node = FlowchartNode.get_root_node(id)
```

Comment:
Do you need this helper to get the root node? Since we set `root_id` for each Flowchart, maybe you could call `root_node = FlowchartNode.find(self.root_id)`?

Reply:
I agree. I just pushed these changes.